### PR TITLE
New rules for CVE-2023-46214

### PR DESCRIPTION
**Pull request by ts-lbf**
### Summary of the Pull Request

This PR adds the following 2 new rules to detect exploitation attempts of CVE-2023-46214:

- A strict detection based on the PoC made by nathan31337.
- A generic detection derived from the POC and based on the splunk team work. (by @nasbench)

### Changelog

new: Exploitation Attempt Of CVE-2023-46214 Using Public POC Code
new: Potential CVE-2023-46214 Exploitation Attempt

### Example Log Event

N/A

### Fixed Issues

N/A

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

File: rules-emerging-threats/2023/Exploits/CVE-2023-46214/web_cve_2023_46214_shell_upload.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - attack.t1210
+logsource:
+    category: webserver
+detection:
+    selection_method:
+        cs-method: POST
+    selection_uris:
+        cs-uri-query|contains:
+            - /splunkd/__upload/indexing/preview\?output_mode=json\&props.NO_BINARY_CHECK=1&input.path=shell.xsl
+            - /api/search/jobs/[0-9]*/results\?xsl=/opt/splunk/var/run/splunk/dispatch/.*/shell.xsl
```
Comment (frack113) [inline rules-emerging-threats/2023/Exploits/CVE-2023-46214/web_cve_2023_46214_shell_upload.yml:23]: Hi,
It is a regex not a string search.
Should be like :
```yaml
cs-uri-query|contains|all:
    - '/api/search/jobs/'
    - '/results\?xsl=/opt/splunk/var/run/splunk/dispatch/'
    - '/shell.xsl'
```

Comment (ts-lbf) [reply rules-emerging-threats/2023/Exploits/CVE-2023-46214/web_cve_2023_46214_shell_upload.yml]: I see, thanks for pointing it out

**Review by nasbench — APPROVED** [review]

File: rules-emerging-threats/2023/Exploits/CVE-2023-46214/web_cve_2023_46214_rce_splunk_enterprise.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+title: Potential CVE-2023-46214 Exploitation Attempt
+id: 04017cd5-621e-4ec4-a762-1f042fe3d3e5
+related:
+    - id: ba5268de-4dd4-4d5c-8a90-2b5e6dc1aff8
+      type: derived
+status: experimental
+description: |
+    Detects potential explotation of CVE-2023-46214 an Remote code execution (RCE) in Splunk Enterprise through Insecure XML Parsing
```
Comment (phantinuss) [inline rules-emerging-threats/2023/Exploits/CVE-2023-46214/web_cve_2023_46214_rce_splunk_enterprise.yml:8]:
```suggestion
    Detects potential exploitation of CVE-2023-46214, a remote code execution (RCE) in Splunk Enterprise through insecure XML parsing
```

File: rules-emerging-threats/2023/Exploits/CVE-2023-46214/web_cve_2023_46214_rce_splunk_enterprise_poc.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+title: Exploitation Attempt Of CVE-2023-46214 Using Public POC Code
+id: ba5268de-4dd4-4d5c-8a90-2b5e6dc1aff8
+related:
+    - id: 04017cd5-621e-4ec4-a762-1f042fe3d3e5
+      type: derived
+status: experimental
+description: |
+    Detects explotation attempt of CVE-2023-46214 an Remote code execution (RCE) in Splunk Enterprise through Insecure XML Parsing using known public Proof of Concept code
```
Comment (phantinuss) [inline rules-emerging-threats/2023/Exploits/CVE-2023-46214/web_cve_2023_46214_rce_splunk_enterprise_poc.yml:8]:
```suggestion
    Detects exploitation attempt of CVE-2023-46214, a remote code execution (RCE) in Splunk Enterprise through insecure XML parsing using known public proof of concept code
```

**Review by phantinuss — APPROVED** [review]

**Pull request merged**